piccolo/query/methods/insert.py

```diff
@@ -55,21 +74,47 @@ def _raw_response_callback(self, results):
 
     @property
     def default_querystrings(self) -> t.Sequence[QueryString]:
-        base = f'INSERT INTO "{self.table._meta.tablename}"'
+        engine_type = self.engine_type
+        if (
+            engine_type == "sqlite"
+            and self.table._meta.db.get_version_sync() < 3.24
+        ):  # pragma: no cover
+            if self.on_conflict_delegate._on_conflict == OnConflict.do_nothing:
+                base = f'INSERT OR IGNORE INTO "{self.table._meta.tablename}"'
+            else:
```

Comment:
I would explicitly check this against `OnConflict.do_update`, and have the else clause raise a `ValueError`.

Reply:
I need clarification. Which `ValueError` should I raise because the user only has two valid options (nothing or update) and if the user enters a wrong value for `on_conflict` the `AttributeError` is already raised? If no `on_conflict` clause is specified, the standard insert query is executed.

Reply:
The value is meant to be an enum but because it's Python could be anything if the user set it to say a string by accident.

```python
if self.on_conflict_delegate._on_conflict == OnConflict.do_nothing:
    base = f'INSERT OR IGNORE INTO "{self.table._meta.tablename}"'
elif self.on_conflict_delegate._on_conflict == OnConflict.do_update:
    ...
else:
    raise ValueError('Invalid on conflict value')

Reply:
OK. I will change that.